runtime/common/bigint.go

```diff
@@ -42,17 +44,46 @@ func NewBigIntFromAbsoluteValue(gauge MemoryGauge, value *big.Int) *big.Int {
 	return new(big.Int).Abs(value)
 }
 
+const minBigIntLength = 4 * BigIntWordSize
+
 // OverEstimateBigIntFromString is an approximate inverse of `interpreter.OverEstimateBigIntStringLength`.
 // Returns the estimated size in bytes.
-func OverEstimateBigIntFromString(s string) int {
+func OverEstimateBigIntFromString(s string, literalKind IntegerLiteralKind) int {
 	l := len(s)
 
-	// Use 6804/2028 to over estimate log_2(10)
-	bitLen := (l*6804)>>11 + 1
+	// By definition: log_b(v) = log_2(v) / log_2(b)
+	// i.e: log_2(v) = log_b(v) * log_2(b)
+
+	// Each digit in base 'b' requires 'log_2(b)' number of digits to represent in binary.
+	// Therefore, to get the number of digits in base 'b', we could multiply by 'log_2(b)'.
```

Comment:
Since `log_2(b)` is not necessarily an integer, I suggest to change the wording a bit. 

```suggestion
        // an integer `v` requires `ceiling(log_b(v))` digits in base `b`, and `ceiling(log_2(v))` digits in base `2`. 
        
        // By definition: log_b(v) = log_2(v) / log_2(b)
	// i.e: log_2(v) = log_b(v) * log_2(b)
	// We can therefore overestimate `ceiling(log_2(v))` by `ceiling(log_b(v)) * ceiling(log_b(v))`.
```